File: EDUARDO/JAVA/AppCompleto_v12/app/src/main/java/fun/eduardo/funcoes/setup_function_rt_media_l.java
package fun.eduardo.funcoes;

import net.trucomanx.pdsplibj.pdsra.*;

public class setup_function_rt_media_l {

	private double Si[];

	public PdsMatrix g0;

	private int i,L;


	public setup_function_rt_media_l(int L){

		this.Si = new double[3];
		this.g0 = new PdsMatrix(1,3);

		this.i = 0;

		this.L = L;


	}	

	
	//ar: input signal  [1x3]
    //return :::
	//g0: A media dos L primeiros [1x3]

   	public PdsMatrix media_de_l(double ar[]) {   

	
		if(this.i<(this.L))
		{
       
   	        this.Si[0] = this.Si[0] + ar[0];               
			this.Si[1] = this.Si[1] + ar[1];               
			this.Si[2] = this.Si[2] + ar[2];

			this.i = this.i+1;                
        	                  
		}
            
        if(this.i == (this.L))
        {      
            this.g0.SetValue(0,0,this.Si[0]/L);
			this.g0.SetValue(0,1,this.Si[1]/L);
			this.g0.SetValue(0,2,this.Si[2]/L);
                
        }

	
        return(g0);
	}

	//public PdsMatrix get_g0(){
	//	return (g0);
	//}


	

}

File: EDUARDO/Matlab_OO_RT/setup_function_rt_media_l.m
% Classe que tira a media dos L primeiros dados
classdef setup_function_rt_media_l < handle
   properties
        g0
        I
        Si
        L
   end
   methods
        %Construtor inializador do objeto
        % inival: inicia todas as variaveis
        function obj =setup_function_rt_media_l(L)
        
            obj.g0=[0 0 0];
            obj.I=0;
            obj.Si=[0 0 0];
            obj.L=L;
            
        end
        %ar: input signal  [1x3]
        %return :::
        %g0: A media dos L primeiros [1x3]
        function g0 = media_de_l(obj,ar)
            
            if obj.I < obj.L+1
                
                obj.Si = obj.Si + ar;                
                obj.I = obj.I + 1;          
                
            end
            
            if obj.I == obj.L+1
               
                obj.g0 = (obj.Si)/obj.L;
                
            end
        
            g0 = obj.g0;
            
          end

   end
end